libunwind/src/AddressSpace.hpp

```diff
@@ -669,8 +669,32 @@ inline bool LocalAddressSpace::findUnwindSections(pint_t targetAddr,
 
         if (found_obj && found_hdr) {
           CHERI_DBG("found_obj && found_hdr in %s\n", pinfo->dlpi_name);
-          cbdata->sects->dwarf_section_length = object_length;
-          return true;
+
+	  // Find the PT_LOAD containing .eh_frame.
+          for (Elf_Half i = 0; i < pinfo->dlpi_phnum; i++) {
+            const Elf_Phdr *phdr = &pinfo->dlpi_phdr[i];
+            if (phdr->p_type == PT_LOAD) {
+              uintptr_t begin = getPhdrCapability(pinfo, phdr);
```

Comment:
I'm not sure my implementation of getPhdrCapability is quite right but if it works here LGTM.

Reply:
It works currently.  I might look at it it further.  It should probably be using assert_pointer_in_bounds() or some such not just checking the tag.  Hmm, I think more thought is required.